### PR TITLE
fixes a few staging Errors, adds Grills layer to asteroids, fixes plasma generation initialisation

### PR DESCRIPTION
**Pull request by Bod9001**
adds Grills layer to asteroids

I added more logging with inventory interactions

fix an error with number dial

insured stuff removes itself from the update manager properly

fixes plasma generation initialisation

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs [hunk 1]
```diff
@@ -52,14 +52,21 @@ private enum SpriteState
...
+
+		public void Initialise()
+		{
+			registerTile = GetComponent<RegisterTile>();
+			securable = GetComponent<WrenchSecurable>();
+			baseSpriteHandler = GetComponentInChildren<SpriteHandler>();
+			electricalNodeControl = GetComponent<ElectricalNodeControl>();
 			var itemStorage = GetComponent<ItemStorage>();
 			itemSlot = itemStorage.GetIndexedItemSlot(0);
 			securable.OnAnchoredChange.AddListener(OnSecuredChanged);
```
Comment (Aranclanos) [inline UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: you're adding two listeners to this now, it will be called twice

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: I don't understand what's wrong with the way it is already, please explain further what you're trying to accomplish

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: On spawn server isn't called when it's already on the scene I think

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: its called at roundstart

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: tho  there are errors on staging relating to it not be initialised properlybefore being used

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: TryToggleOn  errored  because itemSlot was null

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: Yeah, ItemSlot / ItemStorage stuff are snowflakes and can't be used in Awake. Start might be ok? Would have see how they work.

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Engineering/PowerGenerator.cs:61]: Turns out, that same storage method `GetIndexedItemSlot` is used elsewhere in other monobehaviour `Start()`s so presumably it is ok there.

File: UnityProject/Assets/Scripts/Items/Implants/Organs/Liver.cs [hunk 2]
```diff
@@ -66,70 +66,98 @@ public override void ImplantPeriodicUpdate()
...
-					if (tickPullProcessingAmnt <= 0) break;
+						drawnAmount += amount;
+						tickPullProcessingAmnt -= amount;
+						if (tickPullProcessingAmnt <= 0) break;
+					}
 				}
+
 			}
 
+			List<Tuple<Reagent,float>> Toremove = new List<Tuple<Reagent,float>>();
```
Comment (pyrosshade) [inline UnityProject/Assets/Scripts/Items/Implants/Organs/Liver.cs:94]: this and the extra foreachs you presented aren't needed, that's adding a whole other temporary data structure on top of a temporary data structure. If temp array is having issues being removed from in the foreach, then i'll just change it so the temparray is being cleared afterwards. A bit miffed i wasn't awake to see this pr before it was merged.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Implants/Organs/Liver.cs:94]: wouldn't iterating backwards be good enough to solve the issue?

Comment (pyrosshade) [reply UnityProject/Assets/Scripts/Items/Implants/Organs/Liver.cs:94]: it's not an issue of how it's iterating, just an issue of what temparray is used for. it's simple enough to just clear temparray after the foreachs that read through it, which i'm gonna pull up a pr to do just that.

**Pull request merged**